File: types/import.ts
export type ImportItem = {
  /**
   * 問題文または画像URL
   */
  subjects: string[];

  /**
   * 選択肢
   */
  choices: string[];

  /**
   * コミュニティの解答分布(主要な解答のみ)
   */
  communityVotes: string[];

  /**
   * subjectsに存在する画像URLのインデックス群(省略可能)
   */
  indicateSubjectImgIdxes?: number[];

  /**
   * 各選択肢の文末に存在する画像URL(省略可能、nullの場合は画像が存在しない)
   */
  indicateChoiceImgs?: (string | null)[];

  /**
   * 翻訳不必要な文字列のインデックス群(省略可能)
   */
  escapeTranslatedIdxes?: {
    /**
     * subjects(省略可能)
     */
    subjects?: number[];

    /**
     * choices(省略可能)
     */
    choices?: number[];
  };
};
export type ImportDatabaseData = {
  [testName: string]: ImportItem[];
};
export type ImportData = {
  [courseName: string]: ImportDatabaseData;
};

File: functions/types/import.py
"""Import Data Type Definitions"""

from typing import List, Optional, TypedDict


class EscapeTranslatedIdxes(TypedDict, total=False):
    """
    Type of escapeTranslatedIdxes in the array elements of the import data file
    """

    subjects: Optional[List[int]]
    choices: Optional[List[int]]


class ImportItem(TypedDict):
    """
    Type of array elements in the import data file
    """

    subjects: List[str]
    choices: List[str]
    communityVotes: List[str]
    indicateSubjectImgIdxes: Optional[List[int]]
    indicateChoiceImgs: Optional[List[Optional[str]]]
    escapeTranslatedIdxes: Optional[EscapeTranslatedIdxes]


class ImportDatabaseData(TypedDict):
    """
    Type that aggregates all import data files for each test
    """

    __root__: dict[str, List[ImportItem]]


class ImportData(TypedDict):
    """
    Type that aggregates all import data files for tests by course
    """

    __root__: dict[str, ImportDatabaseData]
